lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -3042,21 +3042,31 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       const bool wasAlreadyInvalid = ED->isInvalid();
       ED->setInvalid();
       if (!extType->hasError() && extType->getAnyNominal()) {
+        auto canExtType = extType->getCanonicalType();
+        if (auto existential = canExtType->getAs<ExistentialType>()) {
+          ED->diagnose(diag::unsupported_existential_extension, extType)
+              .highlight(ED->getExtendedTypeRepr()->getSourceRange());
+          ED->diagnose(diag::invalid_extension_rewrite,
+                       existential->getConstraintType())
+              .fixItReplace(ED->getExtendedTypeRepr()->getSourceRange(),
+                            canExtType->getString());
```

Comment:
Shouldn’t this be getConstraintType().getString()?

Reply:
Oops, yeah